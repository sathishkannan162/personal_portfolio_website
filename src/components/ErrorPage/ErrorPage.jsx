// eslint-disable-next-line no-unused-vars
import React from 'react';
import { useRouteError } from 'react-router-dom';

export default function ErrorPage() {
  const error = useRouteError();
  console.error(error);
  if (!error) {
    return (
      <div data-test="error-page">
        <div data-test="error-message">Sorry, an unexpected error has occurred.</div>
      </div>
    );
  }

  return (
    <div id="error-page" data-test="error-page" >
      <h1 data-test="error-oops">Oops!</h1>
      <p>Sorry, an unexpected error has occurred.</p>
      <p>
        <i data-test="error-message">{error.statusText || error.message}</i>
      </p>
    </div>
  );
}
