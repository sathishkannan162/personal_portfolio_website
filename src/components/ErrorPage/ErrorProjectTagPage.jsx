import Pagecontainer from '../CommonComponents/PageContainer';

export default function ErrorProjectTagPage({ message }) {
  return (
    <Pagecontainer>
      <div id="error-page">
        <h1 data-test="error-oops">Oops!</h1>
        <p>Sorry, an unexpected error has occurred.</p>
        <i data-test="error-message">{message}</i>
      </div>
    </Pagecontainer>
  );
}
