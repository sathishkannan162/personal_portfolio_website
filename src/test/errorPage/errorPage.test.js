// eslint-disable-next-line no-unused-vars
import React from 'react';
import ErrorPage from '../../components/ErrorPage/ErrorPage';
import { render, screen } from '@testing-library/react'

jest.mock('react-router-dom', () => ({
  useRouteError: jest.fn(),
}));

describe('ErrorPage', () => {
  it('renders error page with error message', () => {
    const errorMock = {
      statusText: 'Not Found',
      message: 'Page not found',
    };

    // Mock the useRouteError hook to return the error object
    jest.spyOn(require('react-router-dom'), 'useRouteError').mockReturnValue(errorMock);

    const {asFragment} = render(<ErrorPage />);
    expect(asFragment()).toMatchSnapshot();

    // Check if the error page is rendered
    const errorPage = screen.getByTestId('error-page');
    expect(errorPage).toBeInTheDocument();

    // Check if the error message is rendered with the correct text
    const errorMessage = screen.getByTestId('error-message');
    expect(errorMessage).toHaveTextContent('Not Found');
  });

  it('renders error page with default message when error is undefined', () => {
    // Mock the useRouteError hook to return undefined
    jest.spyOn(require('react-router-dom'), 'useRouteError').mockReturnValue(undefined);

    const {asFragment} = render(<ErrorPage />);
    expect(asFragment()).toMatchSnapshot();

    // Check if the error page is rendered
    const errorPage = screen.getByTestId('error-page');
    expect(errorPage).toBeInTheDocument();

    // Check if the default error message is rendered
    const errorMessage = screen.getByTestId('error-message');
    expect(errorMessage).toHaveTextContent('Sorry, an unexpected error has occurred.');
  });
});
