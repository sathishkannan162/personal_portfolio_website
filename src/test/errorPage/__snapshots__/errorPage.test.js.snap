// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage renders error page with default message when error is undefined 1`] = `
<DocumentFragment>
  <div
    data-test="error-page"
  >
    <div
      data-test="error-message"
    >
      Sorry, an unexpected error has occurred.
    </div>
  </div>
</DocumentFragment>
`;

exports[`ErrorPage renders error page with error message 1`] = `
<DocumentFragment>
  <div
    data-test="error-page"
    id="error-page"
  >
    <h1
      data-test="error-oops"
    >
      Oops!
    </h1>
    <p>
      Sorry, an unexpected error has occurred.
    </p>
    <p>
      <i
        data-test="error-message"
      >
        Not Found
      </i>
    </p>
  </div>
</DocumentFragment>
`;
